Math/NumberTheory/GCD.hs

```diff
@@ -252,6 +261,31 @@ cw16 (W16# x#) (W16# y#) = coprimeWord# x# y#
 cw32 :: Word32 -> Word32 -> Bool
 cw32 (W32# x#) (W32# y#) = coprimeWord# x# y#
 
+
+newtype Coprimes a b = Coprimes { unCoprimes :: Map.Map a b } deriving (Eq, Show)
+
+singleton :: Ord a => a -> b -> Coprimes a b
+singleton a b = Coprimes (Map.singleton a b)
+
+toList :: Coprimes a b -> [(a, b)]
+toList x = Map.assocs $ unCoprimes x
+
+insert :: (Integral a, Bits a, Eq b, Num b) => a -> b -> Coprimes a b -> Coprimes a b
```

Comment:
1. Why do we need `Bits a` constraint here?
2. Please cover implementation of `insert`with some basic tests

Reply:
> Why do we need Bits a constraint here?

`coprime :: (Integral a, Bits a) => a -> a -> Bool` is used in `insert`